frontend/appflowy_flutter/lib/plugins/database_view/grid/presentation/grid_page.dart

```diff
@@ -256,16 +249,21 @@ class _GridRowsState extends State<_GridRows> {
                       .read<GridBloc>()
                       .add(GridEvent.moveRow(fromIndex, toIndex));
                 },
-                itemCount: rowInfos.length,
-                itemBuilder: (BuildContext context, int index) {
-                  final RowInfo rowInfo = rowInfos[index];
-                  return _renderRow(
-                    context,
-                    rowInfo,
-                    index: index,
-                    isSortEnabled: sortState.sortInfos.isNotEmpty,
-                    isFilterEnabled: filterState.filters.isNotEmpty,
-                  );
+                itemCount: rowInfos.length + 1,
+                itemBuilder: (context, index) {
+                  if (index < rowInfos.length) {
+                    final rowInfo = rowInfos[index];
+
+                    return _renderRow(
```

Comment:
If this returns a widget, shouldn't we make another widget instead of a function that returns a widget?

Reply:
That is also my preference. I dislike returning widgets inside methods and functions.

But I didn't want to change it in already working code, since in this use-case it doesn't really have a demerit other than preference..